packages/data-stores/src/queries/use-has-seen-whats-new-modal-query.ts

```diff
@@ -0,0 +1,67 @@
+import { useCallback } from 'react';
+import { useQuery, useQueryClient, useMutation } from 'react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+
+interface HasSeenWhatsNewModal {
+	hasSeenWhatsNewModal: boolean;
+}
+
+interface HasSeenWhatsNewModalResult {
+	has_seen_whats_new_modal: boolean;
+}
+
+interface UpdateError {
+	message: string;
+	error: string;
+}
+
+export const useHasSeenWhatsNewModalQuery = ( siteId: number | null ) => {
+	const queryKey = 'has-seen-whats-new-modal';
+
+	const { data, isLoading } = useQuery< { has_seen_whats_new_modal: boolean } >(
+		queryKey,
+		async () =>
+			await wpcomRequest( {
+				path: `/sites/${ siteId }/block-editor/has-seen-whats-new-modal`,
+				apiNamespace: 'wpcom/v2',
+			} ),
+		{
```

Comment:
I think these `async` and `await` cancel each other. The `wpcomRequest` returns a promise anyway.

```suggestion
		() =>
			wpcomRequest( {
				path: `/sites/${ siteId }/block-editor/has-seen-whats-new-modal`,
				apiNamespace: 'wpcom/v2',
			} ),
		{
```